src/rsa.rs

```diff
@@ -605,4 +605,39 @@ mod tests {
         assert!(key_pair.sign(&RSA_PKCS1_SHA256, &rng, MESSAGE,
                               &mut signature).is_err());
     }
+
+    // Test for `primitive::verify()`. Read public key parts from a file
+    // and use them to verify a signature.
+    #[test]
+    fn test_signature_rsa_primitive_verification() {
+        file_test::run("src/rsa_primitive_verify_tests.txt", |section, test_case| {
+            assert_eq!(section, "");
+
+            // Sign the message "hello, world", using PKCS#1 v1.5 padding and the
+            // SHA256 digest algorithm.
+            const MESSAGE: &'static [u8] = b"hello, world";
+            let rng = rand::SystemRandom::new();
+
+            const PRIVATE_KEY_DER: &'static [u8] =
+                include_bytes!("signature_rsa_example_private_key.der");
+            let key_bytes_der = untrusted::Input::from(PRIVATE_KEY_DER);
+            let key_pair = RSAKeyPair::from_der(key_bytes_der).unwrap();
+
+            // The output buffer is one byte too short.
+            let mut signature = vec![0; key_pair.public_modulus_len()];
+            try!(key_pair.sign(&RSA_PKCS1_SHA256, &rng, MESSAGE,
```

Comment:
Let's not call `sign` here. We have the static test vectors in the data files to avoid this pattern where we use the same code to calculate the test vector that we are then going to test. In particular, RSA verification and RSA signing share a lot of code in the modular exponentiation step (especially on 32-bit platforms) and if there is a bug in modular multiplication or modular exponentiation then it is too likely that the bug would go unnoticed with this style, since the computation of the signature and the computation of the verification of the signature might miscalculate the same way.

Plus, we already have significant performance problems with the tests on ARM where we are right (sometimes past) the timeout that Travis CI imposes, so as the set of test vectors for this grows, calling `sign` for each one would eventually be too wasteful.

The bulk of the test vectors for RSA signature verification came from NIST already in this form (separate `n` and `e` attributes) and I had to DER-encode them. We can't afford to have all the test vectors from NIST go through both this function and the normal `verify` test. But, I would like to see some assurance that all the edge cases tested through both APIs.  Maybe the best way to do that, while minimizing redundant work in the CI, is to document very clearly in `VerificationAlgorithmImpl::verify` that the test coverage for this new API depends on `VerificationAlgorithmImpl::verify` being implemented in terms of it. e.g. a comment like this immediately above the call to `verify_rsa`:

```
// The bulk of test coverage for "signature::primitive::verify_rsa"
// depends on it being called through this function. If we change
// that, then the test coverage for `verify_rsa needs to be
// reconsidered.